_data-prepper/pipelines/configuration/processors/list-to-map.md

```diff
@@ -16,10 +16,12 @@ The following table describes the configuration options used to generate target
 
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
-`key` | Yes | String | The key of the fields to be extracted as keys in the generated mappings.
 `source` | Yes | String | The list of objects with `key` fields to be converted into keys for the generated map.
 `target` | No | String | The target for the generated map. When not specified, the generated map will be placed in the root node.
+`key` | No | String | The key of the fields to be extracted as keys in the generated mappings. Must be specified if `use_source_key` is `false`.
+`use_source_key` | No | Boolean | When `true`, keys in the generated map will use original keys from the source. Default is false.
 `value_key` | No | String | When specified, values given a `value_key` in objects contained in the source list will be extracted and converted into the value specified by this option based on the generated map. When not specified, objects contained in the source list retain their original value when mapped.
+`extract_value` | No | Boolean | When `true`, values in objects from the source list will be extracted and put into the generated map. Otherwise, objects from the source list will be put into the generated map. Default is `false`
```

Comment:
Do objects end up in the `generated map` no matter what, despite the value of the boolean? What's the difference between `true` and `false`, when the values are extracted versus place directly into the map?

Reply:
If this boolean is false (no extraction), the objects end up in the generated map; if false, the values from the object will be extracted, so only the values end up in the generated map.

For example, if my input is:
```json
"mylist": [
    {
      "name": "a",
      "value": "val-a"
    },
    {
      "name": "b",
      "value": "val-b"
    }
]
```
and the processor config is:
```yaml
    - list_to_map:
        source: "mylist"
        use_source_key: true
        extract_value: false
```
I will get these output added. The objects end up in the generated map as is:
```json
{
  "name": ["a", "b"],
  "value": [
      {
        "name": "a",
        "value": "val-a"
      },
      {
        "name": "b",
        "value": "val-b"
      }
    ]
}
```

But if extract_value is set to true, I will get the values extracted from original objects like this:
```json
{
  "name": ["a", "b"],
  "value": ["val-a", "val-b"]
}
```